cmd/spiffe-helper/config/config.go

```diff
@@ -164,16 +172,49 @@ func (c *Config) ValidateConfig(log logrus.FieldLogger) error {
 		return errors.New("at least one of the sets ('svid_file_name', 'svid_key_file_name', 'svid_bundle_file_name'), 'jwt_svids', or 'jwt_bundle_file_name' must be fully specified")
 	}
 
+	if c.CertFileMode < 0 {
+		return errors.New("cert file mode must be positive")
+	}
+	if c.KeyFileMode < 0 {
+		return errors.New("key file mode must be positive")
+	}
+	if c.JwtSVIDFileMode < 0 {
+		return errors.New("jwt bundle file mode must be positive")
+	}
+	if c.JwtSVIDFileMode < 0 {
+		return errors.New("jwt svid file mode must be positive")
+	}
+
 	return nil
 }
 
 func NewSidecarConfig(config *Config, log logrus.FieldLogger) *sidecar.Config {
+	certFileMode := defaultCertFileMode
+	if config.CertFileMode > 0 {
+		certFileMode = os.FileMode(config.CertFileMode) //nolint:gosec,G115
+	}
+	keyFileMode := defaultKeyFileMode
+	if config.KeyFileMode > 0 {
+		certFileMode = os.FileMode(config.KeyFileMode) //nolint:gosec,G115
+	}
+	jwtBundleFileMode := defaultJWTBundleFileMode
+	if config.JwtBundleFileMode > 0 {
+		certFileMode = os.FileMode(config.JwtBundleFileMode) //nolint:gosec,G115
+	}
+	jwtSVIDFileMode := defaultJWTSVIDFileMode
+	if config.JwtSVIDFileMode > 0 {
+		certFileMode = os.FileMode(config.JwtSVIDFileMode) //nolint:gosec,G115
+	}
```

Comment:
Can we move this section up in to ParseConfig() where the rest of the defaults are set?

Reply:
I don't see any defaults being set in `ParseConfig()`. This is that entire function
```go
func ParseConfig(file string) (*Config, error) {
	// Read HCL file
	dat, err := os.ReadFile(file)
	if err != nil {
		return nil, err
	}

	// Parse HCL
	config := new(Config)
	if err := hcl.Decode(config, string(dat)); err != nil {
		return nil, err
	}

	return config, nil
}
```

Reply:
Sorry meant to say `ValidateConfig()`

Reply:
Ah, Ok. Moved.

Reply:
Thanks